modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2404,14 +2404,18 @@ def _compute_duplicated(df):
 
         new_index = self._modin_frame._index_cache
         new_columns = [MODIN_UNNAMED_SERIES_LABEL]
-        if len(self.columns) > 1:
-            # if the number of columns we are checking for duplicates is larger than 1,
-            # we must hash them to generate a single value that can be compared across rows.
+        new_row_lengths = self._modin_frame._row_lengths_cache
+        new_column_widths = [1]
+        if len(self._modin_frame.column_widths) > 1:
```

Comment:
Why are you using lengths instead of `columns`? Is there any other reason besides reducing the amount of data transferred?

Reply:
not actual

Reply:
This is perf related. We can apply `duplicated` for a single column partition right away.